osbrain/tests/test_helper.py

```diff
@@ -108,11 +108,11 @@ def test_logger_received_count(agent_logger):
     (42, None, None, 14, False),
     (42, None, None, 42, True),
```

Comment:
Would be great to add some tests that actually verify the result when setting multiple attributes. In example:

- length + data: where data is not in the attribute.
- length + data: where length is too short.
- length + data: a positive case